src/app/components/harvest/screens/metadata-review/metadata-review.component.spec.ts

```diff
@@ -120,4 +182,172 @@ describe("MetaDatReviewComponent", () => {
+      // we can compare these two objects values without the injector class, but still retain verbose failed exceptions & output
+      expect(JSON.stringify(mappings)).toEqual(JSON.stringify(expectedMappings));
+    });
+
+    // modifying the mappings for path "A/aa" will trigger the mocked method above to asset if the correct parameters are called
+    clickEditMappingButton(2);
+    toggleHarvestMappingRecursive(0);
+    expect(harvestService.updateMappings).toHaveBeenCalledTimes(1);
+
+    // close folder A in the directory tree
+    getHarvestItemsSpy.and.callThrough();
+    getHarvestItemsSpy.and.resolveTo(folderStructureFactory());
+    clickFolder("A");
+
+    // open folder B in the directory tree
+    const folderBStructure: HarvestItem[] = folderStructureFactory(["B/ba", "B/bb"]);
+    getHarvestItemsSpy.and.callThrough();
+    getHarvestItemsSpy.and.resolveTo(folderBStructure);
+    clickFolder("B");
+
+    // assert mappings for path "A/aa" have been retained once it is no longer visible in the DOM and mappings have been applied to "B/bb"
+    harvestService.updateMappings.and.callFake((model: Harvest, mappings: HarvestMapping[]) => {
+      const expectedMappings = [
+        new HarvestMapping({
+          path: "A/aa",
+          recursive: false,
+          siteId: null,
+          utcOffset: null
+        }),
+        new HarvestMapping({
+          path: "B/bb",
+          recursive: false,
+          siteId: null,
+          utcOffset: null
+        })
+      ];
+
+      expect(model).toEqual(spec.component.harvest);
+      expect(JSON.stringify(mappings)).toEqual(JSON.stringify(expectedMappings));
+    });
+
+    clickEditMappingButton(4);
+    toggleHarvestMappingRecursive(0);
+    expect(harvestService.updateMappings).toHaveBeenCalledTimes(2);
+
+    flush();
+    discardPeriodicTasks();
+  }));
+
+  fit("should retain mappings assigned in previous stages after change in mapping through interface", fakeAsync(() => {
+    setup();
+
+    const mapping: HarvestMapping[] = [
+      new HarvestMapping({
+        path: "A",
+        recursive: true,
+        siteId: 543,
+        utcOffset: "+11:00"
+      }),
+      new HarvestMapping({
+        path: "B",
+        recursive: true,
+        siteId: null,
+        utcOffset: "+10:00"
+      }, spec.component.injector),
+      new HarvestMapping({
+        path: "C",
+        recursive: true,
+        siteId: 1234,
+        utcOffset: "+10:00"
+      }),
+      new HarvestMapping({
+        path: "C/ca",
+        recursive: false,
+        siteId: null,
+        utcOffset: "-08:00"
+      })
+    ];
+
+    // when harvest item rows are organically generated, they will always include an injector services
+    // as part of the association decorator, it requires this injector service, therefore, we have to add one manually
+    Object.defineProperty(mapping[1], "injector", { value: spec.component });
```

Comment:
This is obviously not ideal, I'm working on a more correct solution to the injector requirement for
https://github.com/QutEcoacoustics/workbench-client/blob/master/src/app/models/AssociationDecorators.ts#L179
>
https://github.com/QutEcoacoustics/workbench-client/blob/master/src/app/components/harvest/components/metadata-review/folder-row.component.ts#L56